via/services/http.py

```diff
@@ -0,0 +1,134 @@
+        self._session = _session or requests.Session()
+
+    def get(self, *args, **kwargs):
+        return self.request("GET", *args, **kwargs)
+
+    def put(self, *args, **kwargs):
+        return self.request("PUT", *args, **kwargs)
+
+    def post(self, *args, **kwargs):
+        return self.request("POST", *args, **kwargs)
+
+    def patch(self, *args, **kwargs):
+        return self.request("PATCH", *args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        return self.request("DELETE", *args, **kwargs)
+
+    def request(self, method, url, timeout=(10, 10), **kwargs):
+        """Send a request with `requests` and return the requests.Response object.
+
+        :param method: The HTTP method to use, one of "GET", "PUT", "POST",
+            "PATCH", "DELETE", "OPTIONS" or "HEAD"
+
+        :param url: The URL to request
+
+        :param timeout: How long (in seconds) to wait before raising an error.
+
+            This can be a (connect_timeout, read_timeout) 2-tuple or it can be
+            a single float that will be used as both the connect and read
+            timeout.
+
+            Good practice is to set this to slightly larger than a multiple of
+            3, which is the default TCP packet retransmission window. See:
+            https://docs.python-requests.org/en/master/user/advanced/#timeouts
+
+            Note that the read_timeout is *not* a time limit on the entire
+            response download. It's a time limit on how long to wait *between
+            bytes from the server*. The entire download can take much longer.
+
+        :param kwargs: Any other keyword arguments will be passed directly to
+            requests.Session().request():
+            https://docs.python-requests.org/en/latest/api/#requests.Session.request
+
+        :raise request.exceptions are mapped to via's exception on DEFAULT_ERROR_MAP
+
+        :raise UnhandledUpstreamException: For any non mapped exception raised by requests
+
+        :returns: a request.Response
+        """
+        response = None
+
+        try:  # pylint:disable=too-many-try-statements
+            response = self._session.request(
+                method,
+                url,
+                timeout=timeout,
+                **kwargs,
+            )
+            response.raise_for_status()
+        except RequestException as err:
+            TargetClass = DEFAULT_ERROR_MAP.get(  # pylint:disable=invalid-name
+                err.__class__, UnhandledUpstreamException
+            )
+            raise TargetClass(
+                message=err.args[0] if err.args else None,
+                requests_err=err,
+            ) from err
+
+        return response
+
+    def stream(self, url, method="GET", **kwargs):
+        response = self.request(method=method, url=url, stream=True, **kwargs)
+
+        yield from self.reify_first(self.stream_bytes(response))
+
+    @staticmethod
+    def reify_first(iterable):
```

Comment:
These two are `staticmethod` as they are used on the google side. I don't think that's wrong, if the google side started using `.stream()` I would only change them to be private.